File: week16/mk/maximum_odd_binary_number.go
// 0ms, 3.89mb
package main

import "fmt"

func main() {
	fmt.Println(maximumOddBinaryNumber("010"))
}

func maximumOddBinaryNumber(s string) string {
	oneCount := 0
	zeroCount := 0
	for i := 0; i < len(s); i++ {
		if s[i] == '1' {
			oneCount++
		} else {
			zeroCount++
		}
	}

	result := ""
	if oneCount == 1 {
		for i := 0; i < zeroCount; i++ {
			result += "0"
		}
		return result + "1"
	}

	for i := 0; i < oneCount-1; i++ {
		result += "1"
	}
	for i := 0; i < zeroCount; i++ {
		result += "0"
	}
	return result + "1"
}

File: S02/week16/sejun/maximum-odd-binary-number.java
// 2ms Beats 85.34% of users with Java
// 43.45MB Beats 48.88% of users with Java

class Solution {
		public String maximumOddBinaryNumber(String s) {
			int oneCount = 0;
			for (char sChar : s.toCharArray()) {
				if (sChar == '1') {
					oneCount++;
				}
			}
			StringBuilder result = new StringBuilder();
			for (int i = 0; i < s.length()-1; i++) {
				if (oneCount > 1) {
					result.append("1");
					oneCount--;
				} else {
					result.append("0");
				}
			}
			result.append("1");

			return result.toString();
		}
}
